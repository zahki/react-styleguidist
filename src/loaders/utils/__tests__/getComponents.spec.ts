import path from 'path';
import identity from 'lodash/identity';
import getComponents from '../getComponents';

it('getComponents() should return an object for components', () => {
	const result = getComponents(['Foo.js', 'Bar.js'], {
		configDir: path.resolve(__dirname, '../../../test'),
		getExampleFilename: identity,
		getComponentPathLine: identity,
	} as any);

	expect(result).toMatchSnapshot();
});
