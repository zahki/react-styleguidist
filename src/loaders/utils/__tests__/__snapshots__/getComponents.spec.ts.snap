// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponents() should return an object for components 1`] = `
Array [
  Object {
    "filepath": "../../Foo.js",
    "hasExamples": false,
    "metadata": Object {},
    "module": Object {
      "require": "Foo.js",
    },
    "pathLine": "../../Foo.js",
    "props": Object {
      "require": "!!~/src/loaders/props-loader.js!Foo.js",
    },
    "slug": "foo",
  },
  Object {
    "filepath": "../../Bar.js",
    "hasExamples": false,
    "metadata": Object {},
    "module": Object {
      "require": "Bar.js",
    },
    "pathLine": "../../Bar.js",
    "props": Object {
      "require": "!!~/src/loaders/props-loader.js!Bar.js",
    },
    "slug": "bar",
  },
]
`;
